tests/unit/storage/test_storage.py

```diff
@@ -1,26 +1,30 @@
 import unittest
-from flexmock import flexmock
+from unittest.mock import Mock
+from unittest.mock import call
 import tuned.storage
 
 class StorageStorageTestCase(unittest.TestCase):
 	def test_set(self):
-		mock_provider = flexmock()
+		mock_provider = Mock()
 		factory = tuned.storage.Factory(mock_provider)
 		storage = factory.create("foo")
 
-		mock_provider.should_receive("set").with_args("foo", "optname", "optval").once
 		storage.set("optname", "optval")
+		mock_provider.set.assert_called_once_with("foo", "optname", "optval")
 
 	def test_get(self):
-		mock_provider = flexmock()
+		mock_provider = Mock()
 		factory = tuned.storage.Factory(mock_provider)
 		storage = factory.create("foo")
 
-		mock_provider.should_receive("get").with_args("foo", "optname", None).and_return(None).once.ordered
-		mock_provider.should_receive("get").with_args("foo", "optname", "defval").and_return("defval").once.ordered
-		mock_provider.should_receive("get").with_args("foo", "existing", None).and_return("somevalue").once.ordered
+		storage.get("optname")
+		storage.get("optname", "defval")
+		storage.get("existing")
 
-		self.assertIsNone(storage.get("optname"))
-		self.assertEqual("defval", storage.get("optname", "defval"))
-		self.assertEqual("somevalue", storage.get("existing"))
+		calls = [
+				 call("foo", "optname", None),
+				 call("foo", "optname", "defval"),
+				 call("foo", "existing", None)
+				]
 
+		mock_provider.get.assert_has_calls(calls)
```

Comment:
This no longer checks the return values, right? Is there a way to fix that?

Reply:
This test was badly written. 3 removed calls did the following: they created function `get` and recorded that it should be called 3 times with some arguments and then return None, "defval" and "somevalue". So the assertions were useless because mocked method always returned what the 3 removed calls told it to return.
I do not think that checking the return value here is necessary because it's sufficient implementation would be to set return value of `get` to some value and assert that the same value is returned.

Reply:
I think the point is that the assertions call `storage.get()`, not `mock_provider.get()`. I think the point of the test is to check that `storage.get` correctly passes its arguments to `mock_provider.get()` and then passes back the return value. Right now, the test doesn't check the second part - passing back the return value.

Reply:
I think we can achieve what we want using the following patch. There might be a better way to write it, but this is the best I was able to come up with and it should work.
```
diff --git a/tests/unit/storage/test_storage.py b/tests/unit/storage/test_storage.py
index ca6eef8..2d2709a 100644
--- a/tests/unit/storage/test_storage.py
+++ b/tests/unit/storage/test_storage.py
@@ -14,12 +14,13 @@ class StorageStorageTestCase(unittest.TestCase):
 
 	def test_get(self):
 		mock_provider = Mock()
+		mock_provider.get.side_effect = [ None, "defval", "somevalue" ]
 		factory = tuned.storage.Factory(mock_provider)
 		storage = factory.create("foo")
 
-		storage.get("optname")
-		storage.get("optname", "defval")
-		storage.get("existing")
+		self.assertEqual(None, storage.get("optname"))
+		self.assertEqual("defval", storage.get("optname", "defval"))
+		self.assertEqual("somevalue", storage.get("existing"))
 
 		calls = [
 				 call("foo", "optname", None),
```

Reply:
(Inspired by https://stackoverflow.com/a/24897297)

Reply:
Fixed.